### PR TITLE
LPS-133729 Create new storage adapter related to Object product

### PR DESCRIPTION
**Pull request by gabrielwas**
Related Issue: https://issues.liferay.com/browse/LPS-133729

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hey @gabrielwas, can you please fix the source formatter issues and resend? Thanks

File: modules/apps/object-forms/object-forms-storage/src/main/java/com/liferay/object/forms/storage/internal/model/listener/DDMStructureModelListener.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+		objectFields.add(_createObjectField("ddmStorageId", "Long"));
+
+		try {
+			_objectDefinitionLocalService.addObjectDefinition(
+				ddmStructure.getUserId(),
+				"Structure" + ddmStructure.getStructureId(), objectFields);
+		}
+		catch (PortalException portalException) {
+			_log.error("Unable to add object definition", portalException);
```
Comment (thiago-farias) [inline modules/apps/object-forms/object-forms-storage/src/main/java/com/liferay/object/forms/storage/internal/model/listener/DDMStructureModelListener.java:70]: Should it raise `ModelListenerException`?

File: modules/apps/object-forms/object-forms-storage/src/main/java/com/liferay/object/forms/storage/internal/storage/ObjectsDDMStorageAdapter.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+			objectEntry = _objectEntryLocalService.getObjectEntry(
+				ddmStorageAdapterDeleteRequest.getPrimaryKey());
+
+			Map<String, Serializable> values = objectEntry.getValues();
+
+			_ddmFieldLocalService.deleteDDMFormValues(
+				(Long)values.get("ddmStorageId"));
+
+			_objectEntryLocalService.deleteObjectEntry(
+				objectEntry.getObjectEntryId());
```
Comment (thiago-farias) [inline modules/apps/object-forms/object-forms-storage/src/main/java/com/liferay/object/forms/storage/internal/storage/ObjectsDDMStorageAdapter.java:78]: These instruction should be inside a transaction, or you should simulate it. If it raises an error on line 77, the DDMFormValue is gone, but the related ObjectEntry is still there. It shouldn't occurs.

Comment (gabrielwas) [reply modules/apps/object-forms/object-forms-storage/src/main/java/com/liferay/object/forms/storage/internal/storage/ObjectsDDMStorageAdapter.java:78]: I think when we have a StorageException, the operation rollbacks the changes. But I can test it, of course.

**Comment by gabrielwas** [comment]
ci:open

**Pull request closed without merging**